### PR TITLE
[docs] Add new pages for release versioning and supported releases

### PR DESCRIPTION
**Pull request by stevebang**
This PR includes the following, based on discussions with @chirag-yb and @rkarthik:
- Release versioning page
- Supported and planned releases page
- Notes added to release notes for 2.3, 2.2.2, and 2.2, with links to these pages.
- Note added to listing of earlier releases.

### CODE REVIEW

File: docs/content/latest/releases/releases-overview.md
Comment (chirag-yb) [inline docs/content/latest/releases/releases-overview.md:41]: @stevebang I wouldn't call "changes that may break backward compatibility" anywhere instead say it's not recommended for any production use cases and it's subjected to changes. This will need changes in multiple places

File: docs/content/latest/releases/versioning.md
Comment (chirag-yb) [inline docs/content/latest/releases/versioning.md:31]: @stevebang This is not correct "since the last stable release and may break backward compatibility"

File: docs/content/latest/releases/releases-overview.md
Comment (stevebang) [reply docs/content/latest/releases/releases-overview.md]: Updated all instances to remove that phrase.

File: docs/content/latest/releases/versioning.md
Comment (stevebang) [reply docs/content/latest/releases/versioning.md]: Fixed.

**Review by chirag-yb — CHANGES_REQUESTED** [review]
@stevebang Could you address two comments and resend it?

File: docs/content/latest/releases/releases-overview.md
Comment (chirag-yb) [inline docs/content/latest/releases/releases-overview.md:36]: @stevebang Need to confirm if 1.3 and 2.0 are still supported ? if so, until when? This is going to be written in our support policy 

*****
  Release Support. 
(a) For the maintenance support period of at least 1 year from the major release date, Yugabyte will provide Updates for such release.

(b) Following the maintenance support period, Yugabyte will provide Updates for at least an additional 180 days subject to the following guidelines:
Updates and Upgrades will not be made to the major release.
Yugabyte will direct Customers to existing Updates and workarounds applicable to the reported case.
Yugabyte may direct Customers to Upgrade to a current release if a workaround does not exist.

(c)  Yugabyte will post publicly on its website a notice of End of Life (EOL) for the affected Software and the timeline for discontinuing Support Services. Yugabyte has no obligation to provide Support Services for Software after EOL.

Current Supported Software Releases are listed on Yugabyte’s website.
*****

**Review by chirag-yb — COMMENTED** [review]
@stevebang Need to confirm 1.3 and 2.0 are still supported

File: docs/content/latest/releases/releases-overview.md
Comment (stevebang) [reply docs/content/latest/releases/releases-overview.md:36]: Based on the policy statement above, both of those are still supported. 1.3 would fall under the post-maintenance support time period. I'll keep my eye out for a link to the support policy and add that when it is available.

**Review by chirag-yb — APPROVED** [review]
@stevebang could you check my last comment? Otherwise, it's good to go from my end. 
Waiting for Karthik's approval

File: docs/content/latest/releases/releases-overview.md
Comment (stevebang) [reply docs/content/latest/releases/releases-overview.md:57]: Check the Google Doc — I think I have it correct.

File: docs/content/latest/releases/releases-overview.md
Comment (rkarthik007) [inline docs/content/latest/releases/releases-overview.md:21]: My preference is to combine this table and the one in section below. Just add a column that outlines the recommended latest and stable release. That way, there is one table. This fragmentation is confusing.

Also, a couple of lines below that table, helping the end user quickly decide which release they should pick would be great.

File: docs/content/latest/releases/releases-overview.md
Comment (rkarthik007) [inline docs/content/latest/releases/releases-overview.md:56]: These should be flipped. We will put out the 2.4 release in Jan.

**Review by rkarthik007 — COMMENTED** [review]
Some comments... should we do a live session?

File: docs/content/latest/releases/releases-overview.md
Comment (chirag-yb) [reply docs/content/latest/releases/releases-overview.md:21]: @rkarthik007 do we want to advertise EOL & maintenance support dates for the latest release? I would imagine NO. 
I think adding the latest and future scheduled releases in the same table as supported stable releases are confusing. Can we keep the way it is for now and roll with it?

Comment (stevebang) [reply docs/content/latest/releases/releases-overview.md]: The stable and latest releases releases are combined into one section and table.

File: docs/content/latest/releases/releases-overview.md
Comment (stevebang) [reply docs/content/latest/releases/releases-overview.md]: Dates updated based on discussion with Chirag and comment by @rkarthik007

**Review by chirag-yb — APPROVED** [review]
looks good to me

**Comment by stevebang** [comment]
Moved content from this PR into #5711. Made it much easier to resolve odd merge conflicts.

File: docs/content/latest/releases/releases-overview.md
Comment (chirag-yb) [inline docs/content/latest/releases/releases-overview.md:42]: @stevebang Minor changes on this section
1. Can we reword it to "Upcoming releases"
2. Can we flip the order of version and planned release columns and sort it by the release date?

File: docs/content/latest/releases/releases-overview.md
Comment (chirag-yb) [inline docs/content/latest/releases/releases-overview.md:32]: @stevebang It would be nice to have the status (Latest (development and testing), Current Stable (recommended for production), Stable in parenthesis next to the release series instead of a separate column.

File: docs/content/latest/releases/versioning.md
Comment (chirag-yb) [inline docs/content/latest/releases/versioning.md:25]: @stevebang Lets remove this `that do not break backwards compatibility`

File: docs/content/latest/releases/releases-overview.md
Comment (stevebang) [reply docs/content/latest/releases/releases-overview.md:42]: done

File: docs/content/latest/releases/releases-overview.md
Comment (stevebang) [reply docs/content/latest/releases/releases-overview.md:32]: done

File: docs/content/latest/releases/versioning.md
Comment (stevebang) [reply docs/content/latest/releases/versioning.md:25]: Reworded before your comment, so take a look at what's there now. If needed, I can change it.

**Pull request closed without merging**
